### PR TITLE
[#176013578] refactor pipeline

### PR DESCRIPTION
**Pull request by gquadrati**
- added ./devops folder wirh code-review and deploy pipelines, following the [guidelines](https://pagopa.github.io/io-docs/io-handbook/cicd-setup)
- removed old .circleci/ and azure-pipelines.yml
- switched from release-it to auto-changelog
- created pipelines into [azure devops project](https://dev.azure.com/pagopa-io/io-spid-commons/)

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      - minor
+      - patch
+    default: minor
+
+# Only manual activations are intended
+trigger: none
+pr: none
+
+pool:
+  vmImage: 'ubuntu-latest'
```
Comment (gunzip) [inline .devops/deploy-pipelines.yml:33]: when publishing packages we must use win32 build

Comment (gquadrati) [reply .devops/deploy-pipelines.yml:33]: Actually we setup the same for all the packages, like [io-functions-commons](https://github.com/pagopa/io-functions-commons/blob/1af5e8cdd8cd53a329908be0cd8e037425dd24c0/.devops/deploy-pipelines.yml#L29) and [io-ts-commons](https://github.com/pagopa/io-ts-commons/blob/a81efc9983083688e2d9621a6eca99ec822ef9bb/.devops/deploy-pipelines.yml#L30)

What should I do, change all of them?

**Comment by gquadrati** [comment]
@balanza I setup the pipelines in Azure, but I'm not able to trigger them with a PR.. t looks like the same problem we had yersterday..

File: .devops/deploy-pipelines.yml (hunk 1 above)
Comment (gunzip) [reply .devops/deploy-pipelines.yml:33]: nope, sorry I was wrong it's ok as is (moreover, this ib is used inside appbackend which is deployed on linux machines)

Comment (gquadrati) [reply .devops/deploy-pipelines.yml:33]: good :)

File: README.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+[![Build Status](https://dev.azure.com/pagopa-io/io-spid-commons/_apis/build/status/pagopa.io-spid-commons.code-review?branchName=176013578-refactor-pipeline)](https://dev.azure.com/pagopa-io/io-spid-commons/_build/latest?definitionId=99&branchName=176013578-refactor-pipeline)
```
Comment (balanza) [inline README.md:1]: I like the idea, but as it's configured it's showing no interesting information. This badge is meant to show if master is shippable (meaning: it builds and tests are ok). No great value for as, as we perform all tests we have everytime we merge.

If we had long-running integration/e2e tests that we run **after** the merge, that would really make more sense.

**Review by balanza — APPROVED** [review]
LGTM

File: README.md (hunk 2 above)
Comment (gquadrati) [reply README.md:1]: I'll remove it.. actually I need to remove also from io-ts-commons..

**Review by balanza — APPROVED** [review]

**Pull request merged**